_assets/js/templates/badges/badgeLogic.js

```diff
@@ -0,0 +1,24 @@
+import { replaceState } from '../../utils/replaceHistory';
+import toggleSelectorState from '../../utils/taSelectors/updateTASelectors';
+import toggleVisibility from '../../utils/taSelectors/toggleVisibility';
+import totalResults from '../search/totalResultsTemplate';
+
+export default function badgeLogic(element, state) {
```

Comment:
This checks to see if the badge is in the DOM, and if not, it attaches the badge, and then adds an event listener. I'm thinking of calling it 'attachBadge' because that's what it does and why we call it.